### PR TITLE
feat: catch error on mongo connection

### PR DESCRIPTION
**Pull request by Sanix-Darker**
## WHAT

- catch errors on awsathena when getting an attribute from NoneType
- catch ConfigurationError on mongo-connector

### CODE REVIEW

File: tests/docker-compose.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
+---
+
```
Comment (lukapeschke) [inline tests/docker-compose.yml:2]: why is this needed ?

File: toucan_connectors/mongo/mongo_connector.py [hunk 2]
```diff
@@ -162,12 +162,19 @@ def get_status(self) -> ConnectorStatus:
...
+                client.server_info()
+            except pymongo.errors.ServerSelectionTimeoutError as e:
+                return ConnectorStatus(
+                    status=False, details=self._get_details(2, False), error=str(e)
+                )
+            except pymongo.errors.OperationFailure as e:
+                return ConnectorStatus(
+                    status=False, details=self._get_details(3, False), error=str(e)
+                )
+        except pymongo.errors.ConfigurationError as e:
```
Comment (lukapeschke) [inline toucan_connectors/mongo/mongo_connector.py:177]: could you please unnest the `try` blocks ?

File: toucan_connectors/awsathena/awsathena_connector.py [hunk 3]
```diff
@@ -60,13 +60,19 @@ class AwsathenaConnector(ToucanConnector):
...
+            raise AttributeError(f'Error: failed to get your aws_secret_access_key, {excp}')
+        else:
+            boto_session = boto3.Session(
+                aws_access_key_id=self.aws_access_key_id,
+                # This is required because this gets appended by boto3
+                # internally, and a SecretStr can't be appended to an str
+                aws_secret_access_key=aws_secret_access_key,
+                region_name=self.region_name,
+            )
+            return boto_session
```
[the remark this answers is not in the document]
Comment (Sanix-Darker) [reply toucan_connectors/awsathena/awsathena_connector.py:75]: This is the error raised : 
```
AttributeError: 'NoneType' object has no attribute 'get_secret_value'
```
IMO, the message i added seems more explicit

File: tests/docker-compose.yml (hunk 1 above)
Comment (Sanix-Darker) [reply tests/docker-compose.yml:2]: a result of my yarn linter 😅 !

Comment (lukapeschke) [reply tests/docker-compose.yml:2]: Hmmm that is only needed if you want to describe several YAML objects in the same file, such as for a kubernetes manifest... It's not needed here, so I'd keep the file as is

File: toucan_connectors/awsathena/awsathena_connector.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (lukapeschke) [reply toucan_connectors/awsathena/awsathena_connector.py:75]: I agree that the message is better, but I'm not sure it should be possible to be in a state where we do not have a secret key. Do you know why it is the only optional field ?

[the remark this answers is not in the document]
Comment (Sanix-Darker) [reply toucan_connectors/awsathena/awsathena_connector.py:75]: > Do you know why it is the only optional field ?

Nope 🤔 , i removed the "optional" !

File: toucan_connectors/mongo/mongo_connector.py (hunk 2 above)
Comment (Sanix-Darker) [reply toucan_connectors/mongo/mongo_connector.py]: done here : https://github.com/ToucanToco/toucan-connectors/pull/634/commits/c78eac60ab45a0e5ff6db630d67ba233c74491ed

File: tests/docker-compose.yml (hunk 1 above)
Comment (Sanix-Darker) [reply tests/docker-compose.yml]: Thanks, did knew it 👍🏾

File: toucan_connectors/awsathena/awsathena_connector.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (lukapeschke) [reply toucan_connectors/awsathena/awsathena_connector.py:75]: Since it has been removed, the `try/except` block is not needed anymore

[the remark this answers is not in the document]
Comment (Sanix-Darker) [reply toucan_connectors/awsathena/awsathena_connector.py]: [`4af1caf` (#634)](https://github.com/ToucanToco/toucan-connectors/pull/634/commits/4af1caf91c6968e20bbe28be97e149e7b4eacd68)
